diff --git a/bin/winapps b/bin/winapps
index da59492..eea467f 100755
--- a/bin/winapps
+++ b/bin/winapps
@@ -37,7 +37,7 @@ readonly COMPOSE_PATH="${HOME}/.config/winapps/compose.yaml"
 readonly SCRIPT_DIR_PATH="$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd)"

 # OTHER
-readonly VM_NAME="RDPWindows" # FOR 'libvirt' ONLY
+readonly VM_NAME="windows-10" # FOR 'libvirt' ONLY
 readonly CONTAINER_NAME="WinApps" # FOR 'docker' AND 'podman' ONLY
 readonly RDP_PORT=3389
 readonly DOCKER_IP="127.0.0.1"
